src/sql/data.d.ts

```diff
@@ -1410,6 +1410,38 @@ declare module 'data' {
 		postMessage(message: any): Thenable<any>;
 	}
 
+	export interface WebviewWidget {
+
+		/**
+		 * Raised when the webview posts a message.
+		 */
+		readonly onMessage: vscode.Event<any>;
+
+		/**
+		 * Raised when the widget closed.
+		 */
+		readonly onClosed: vscode.Event<any>;
+
+		/**
+		 * Post a message to the widget.
+		 *
+		 * @param message Body of the message.
+		 */
+		postMessage(message: any): Thenable<any>;
+
+		/**
+		 * Contents of the dialog body.
+		 */
+		html: string;
+	}
+
+	export namespace dashboard {
+		/**
+		 * Register a provider for a webview widget
+		 */
+		export function registerDashboardWebviewWidgetProvider(widgetId: string, handler: (webview: WebviewWidget) => void): void;
```

Comment:
minor: maybe don't need to include "Dashboard" in the function name since it's already in the `dashboard` namespace